Plugin/NonBoltOrderPlugin.php

```diff
@@ -149,7 +146,7 @@ public function execute(Observer $observer)
             $response = $result->getResponse();
             if (empty($response)) {
                 $this->metricsClient->processCountMetric("non_bolt_order_creation.failure", 1);
-                return;
+                return $order;
             }
 
             $boltTransactionData = ['bolt_transaction_reference' => $response->reference];
```

Comment:
@nickfiacco-bolt
Is it necessary to store this data with the payment instance info of some other payment method?
I still see it as a potential cause of issues on more than one level.
Could we add a column to the order table directly and store it there?
cc @njfaries @johnmccombs1 @matt-thomason @vitaliyreznikov @mccottry @daisy1754

Reply:
Nice idea @danac-gs !

Reply:
I can do that-- although for the record PayPal uses this field even for non-PayPal orders in some cases, so the precedent is set if we want to use this field. The reason for the original issue was the timing of when the additional information was set.

Reply:
On second thought there's no downside to a new column, so I'll go with that